src/test/synapse.test.ts

```diff
@@ -255,6 +255,57 @@ describe('Synapse', () => {
     })
   })
 
+  describe('Session Keys', () => {
+    let originalFetch: typeof global.fetch
+
+    beforeEach(() => {
+      originalFetch = global.fetch
+      // Default mock for ping validation
+      global.fetch = async (input: string | URL | Request) => {
+        const url = typeof input === 'string' ? input : input instanceof URL ? input.toString() : input.url
+        if (url.includes('/ping')) {
+          return { status: 200, statusText: 'OK' } as any
+        }
+        throw new Error(`Unexpected URL: ${url}`)
+      }
+    })
+
+    afterEach(() => {
+      global.fetch = originalFetch
+    })
+
+    it('should storage.createContext with session key', async () => {
+      const signerAddress = await signer.getAddress()
+      server.use(
+        JSONRPC({
+          ...presets.basic,
+          payments: {
+            ...presets.basic.payments,
+            accounts: (args) => {
+              const token = args[0]
+              const user = args[1]
+              assert.equal(user, signerAddress)
+              assert.equal(token, ADDRESSES.calibration.usdfcToken)
+              return [BigInt(127001), BigInt(0), BigInt(0), BigInt(0)]
+            },
+          },
+        })
+      )
+      const synapse = await Synapse.create({ signer })
+      const sessionKeySigner = new ethers.Wallet(PRIVATE_KEYS.key2, provider)
+      const sessionKey = synapse.setSession(sessionKeySigner)
```

Comment:
Am I right in assuming that `setSession()` is something we expose for advanced and internal use and not something we expect the user to do manually in the normal case and that we'd be able to handle the dirty work of generating a temporary key for some period while the SDK is active? Or do you think we need to make this the user's responsibility to set this up and make it work, we'll just deal with it if they have gone to the trouble?

Reply:
I don't want the sdk to be opinionated on how the key is generated or stored.